src/openrct2/localisation/LocalisationService.cpp

```diff
@@ -99,48 +95,59 @@ void LocalisationService::OpenLanguage(int32_t id)
     }
 
     std::string filename;
-    if (id != LANGUAGE_ENGLISH_UK)
-    {
-        filename = GetLanguagePath(LANGUAGE_ENGLISH_UK);
-        _languageFallback = LanguagePackFactory::FromFile(LANGUAGE_ENGLISH_UK, filename.c_str());
-    }
-
     filename = GetLanguagePath(id);
-    _languageCurrent = LanguagePackFactory::FromFile(id, filename.c_str());
-    if (_languageCurrent != nullptr)
+    auto preferredLanguage = LanguagePackFactory::FromFile(id, filename.c_str());
+    if (preferredLanguage != nullptr)
     {
         _currentLanguage = id;
+        _loadedLanguages.emplace_back(std::move(preferredLanguage));
         TryLoadFonts(*this);
     }
     else
     {
         throw std::runtime_error("Unable to open language " + std::to_string(id));
     }
+
+    const auto fallback = LanguagesDescriptors[id].fallback;
+    if (fallback != LANGUAGE_UNDEFINED)
+    {
+        filename = GetLanguagePath(fallback);
+        _loadedLanguages.emplace_back(LanguagePackFactory::FromFile(fallback, filename.c_str()));
+    }
+
+    if (id != LANGUAGE_ENGLISH_UK)
+    {
+        filename = GetLanguagePath(LANGUAGE_ENGLISH_UK);
+        _loadedLanguages.emplace_back(LanguagePackFactory::FromFile(LANGUAGE_ENGLISH_UK, filename.c_str()));
+    }
 }
 
 void LocalisationService::CloseLanguages()
 {
-    _languageFallback = nullptr;
-    _languageCurrent = nullptr;
+    for (auto& language : _loadedLanguages)
+    {
+        language = nullptr;
+    }
```

Comment:
The `nullifying` should not be needed, I think. Since they are `unique_ptrs`, once the vector stops referencing them they should go out of scope.